### ISSUE TITLE
cannot start Storybook on latest `master` in `react-components`

### ISSUE THREAD

**Issue by layershifter** [microsoft/fluentui#20886]
Repro:
- `git clone https://github.com/microsoft/fluentui.git`
- `cd fluentui`
- `yarn`
- `yarn workspace @fluentui/react-components start`

![image](https://user-images.githubusercontent.com/14183168/144259536-163a7404-4a3f-4392-a5b0-c166d089f64a.png)

**Comment by layershifter** [comment]
TLDR; We have a hack for `babel-loader` ([.storybook/custom-loader.js](https://github.com/microsoft/fluentui/blob/84e70e873508c0a1b061bb401f7143c1bcd44397/.storybook/custom-loader.js)) to exclude `@fluentui/babel-make-styles`, but in reality it does not as it should.

---

We have a hack that relies on `customize` API in `babel-loader` ([Options](https://github.com/babel/babel-loader#options)), the problem that it executed later than it should:

```js
  const { loadPartialConfigAsync = babel.loadPartialConfig } = babel;
  const config = await loadPartialConfigAsync(
    injectCaller(programmaticOptions, this.target),
  );
  if (config) {
    let options = config.options;
    if (overrides && overrides.config) {
      options = await overrides.config.call(this, config, {
```

> https://github.com/babel/babel-loader/blob/052c07a0168a24e2bb15fd8890c9e485c5a3fe97/src/index.js#L163-L170

An error is thrown on a call of `loadPartialConfigAsync()` while our hack will be executed in `overrides.config.call`: our code is executed later than should 💥 I traced it in `@babel/core` itself and the problem happens in `resolveStandardizedName()` (Babel resolves all plugins to matching files):

```js
  try {
    return require.resolve(standardizedName, {
      paths: [dirname],
    });
```

> https://github.com/babel/babel/blob/36a5ac4145be44b5dbbe2e8a70d1d0e7dc32ba43/packages/babel-core/src/config/files/plugins.ts#L95-L105

```bash
PS fluentui> node -e "console.log(require.resolve('@fluentui/babel-make-styles'))"
```
⬇️⬇️⬇️
```
internal/modules/cjs/loader.js:311
      throw err;
      ^

Error: Cannot find module 'fluentui\node_modules\@fluentui\babel-make-styles\lib\index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:303:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:516:18)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:867:27)
    at Function.resolve (internal/modules/cjs/helpers.js:94:19)
    at [eval]:1:21
    at Script.runInThisContext (vm.js:132:18)
    at Object.runInThisContext (vm.js:309:38)
    at internal/process/execution.js:77:19
    at [eval]-wrapper:6:22
    at evalScript (internal/process/execution.js:76:60) {
  code: 'MODULE_NOT_FOUND',
  path: '@fluentui\\babel-make-styles\\package.json',
  requestPath: '@fluentui/babel-make-styles'
}
```

👆 That matches with a console error:

![image](https://user-images.githubusercontent.com/14183168/144420118-85d9ab5c-2814-4244-af2f-c2aa1e2d517e.png)

**Comment by Hotell** [comment]
thanks for investigation. 

would you able to use git bisect to isolate which change introduced this ? previously the override to exclude babe plugins worked so it would be great to be aware what changed.
